### PR TITLE
Improve HyperBand docs

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

I'd like to suggest a few changes to improve the docstring of HyperBand Pruner.

## Description of the changes

- Fix typos
- Use math syntax for the definition of #bracket since the current definition is slightly difficult to read.
- Remove less informative `seealso` because `report` is only used in examples like other pruners and the first `seealso` section of `max_resource` argument.

### CODE REVIEW

File: optuna/pruners/_hyperband.py [hunk 1]
```diff
@@ -48,19 +48,18 @@ class HyperbandPruner(BasePruner):
         at least :math:`4 \\times 10` trials are consumed for startup.
 
     .. note::
-        Hyperband has several :class:`~optuna.pruners.SuccessiveHalvingPruner`. Each
-        :class:`~optuna.pruners.SuccessiveHalvingPruner` is referred as "bracket" in the original
-        paper. The number of brackets is an important factor to control the early stopping behavior
-        of Hyperband and is automatically determined by ``min_resource``, ``max_resource`` and
-        ``reduction_factor`` as
-        `The number of brackets = floor(log_{reduction_factor}(max_resource / min_resource)) + 1`.
+        Hyperband has several :class:`~optuna.pruners.SuccessiveHalvingPruner`\\ s. Each
```
Comment (nzw0301) [inline optuna/pruners/_hyperband.py:51]: add `\\s` after ":class:~optuna.pruners.SuccessiveHalvingPruner"

File: optuna/pruners/_hyperband.py [hunk 2]
```diff
@@ -48,19 +48,18 @@ class HyperbandPruner(BasePruner):
...
+        :class:`~optuna.pruners.SuccessiveHalvingPruner` is referred to as "bracket" in the
```
Comment (nzw0301) [inline optuna/pruners/_hyperband.py:52]: "is referred as" -> "is referred to as"

**Comment by toshihikoyanase** [comment]
@contramundum53 Could you review this PR, please? I think we can merge this PR with a single approval since it updates the document only.

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Pull request merged**